src/diracx/client/aio/_patch.py

```diff
@@ -20,3 +39,207 @@ def patch_sdk():
     you can't accomplish using the techniques described in
     https://aka.ms/azsdk/python/dpcodegen/python/customize
     """
+
+
+class DiracTokenCredential(AsyncTokenCredential):
+    """Tailor get_token() for our context"""
+
+    def __init__(self, token_endpoint, client_id) -> None:
+        self.token_endpoint = token_endpoint
+        self.client_id = client_id
+
+    async def get_token(
+        self,
+        *scopes: str,
+        claims: Optional[str] = None,
+        tenant_id: Optional[str] = None,
+        **kwargs: Any,
+    ) -> AccessToken:
+        """Refresh the access token using the refresh_token flow.
+        :param str scopes: The type of access needed.
+        :keyword str claims: Additional claims required in the token, such as those returned in a resource
+            provider's claims challenge following an authorization failure.
+        :keyword str tenant_id: Optional tenant to include in the token request.
+        :rtype: AccessToken
+        :return: An AccessToken instance containing the token string and its expiration time in Unix time.
+        """
+        response = requests.post(
+            url=self.token_endpoint,
+            data={
+                "client_id": self.client_id,
+                "grant_type": "refresh_token",
+                "refresh_token": kwargs.get("refresh_token"),
+            },
+        )
+
+        if response.status_code != 200:
+            raise RuntimeError(
+                f"An issue occured while refreshing your access token: {response.json()['detail']}"
+            )
+
+        res = response.json()
+        token_response = TokenResponse(
+            access_token=res["access_token"],
+            expires_in=res["expires_in"],
+            state=res["state"],
+            refresh_token=res.get("refresh_token"),
+            token_type=res.get("token_type"),
+        )
+
+        write_credentials(token_response)
+        credentials = json.loads(CREDENTIALS_PATH.read_text())
+        return AccessToken(
+            credentials.get("access_token"), credentials.get("expires_on")
+        )
+
+    async def close(self) -> None:
```

Comment:
I added a docstring to explain, but basically because `AsyncTokenCredential` is a Python protocol and we need to provide an implementation of the methods.